### PR TITLE
CORE-1189: Extend WKTransferSubmitError

### PR DESCRIPTION
**Pull request by EBGToo**
This extends the public interface of `WKTransferSubmitErrorType` and `WKTransferIncludeStatusType` which are used in `WKTransferState`.

Also extends `WKClientErrorType` and the Swift/Java interface for SystemClientError and SystemClientSubmitError to determine error states from Blockset queries.  The query error responses are 'mapped' to the `WKTransfer*` errors.

### CODE REVIEW

**Review by bgoring — APPROVED** [review]

File: WalletKitCore/src/walletkit/WKClientP.h [hunk 1]
```diff
@@ -28,6 +28,19 @@
...
 
+// MARK: - Error
+
+struct WKClientErrorRecord {
+    WKClientErrorType type;
+    char *details;
+    union {
+        WKTransferSubmitErrorType submitErrorType;
+    } u;
+};
```
[the remark this answers is not in the document]
Comment (EBGToo) [reply WalletKitCore/src/walletkit/WKClientP.h:39]: Yes.

File: WalletKitJava/WalletKit/src/main/java/com/blockset/walletkit/TransferSubmitError.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.blockset.walletkit;
```
[the remark this answers is not in the document]
Comment (EBGToo) [reply WalletKitJava/WalletKit/src/main/java/com/blockset/walletkit/TransferSubmitError.java:1]: There isn't an 'errors' package under com.blockset.walletkit.  Will leave at the same level as 'TransferIncludeStatus'

**Comment by EBGToo** [comment]
@DrewCarlson, @brd-jwheeler Marking this ready to review.  I've not seen data from Blockset yet; if issues arise we'll fix as they occur.

Small App changes are required related to `Transaction` confirmation and submission.  They'll show as compiler errors upon App compilation and adjustment should be obvious.

**Comment by EBGToo** [comment]
Tested this on iOS and Android by injecting Ethereum Transaction Errors (zeroed out the nonce, gas Price to zero, gas Limit to zero, signature vrs.s to zero, and initial 4 bytes in the serialization to zero).  Expected Blockset status to reflect the errors.  Deployed an updated Blockset to 'pre-prod'; pointed `query.createTransaction` (aka 'submit') to 'pre-prod' and then created flawed transactions.

All errors were reported correctly, from WalletKit and then through iOS/Android... with the exception of `gas Price to zero` - apparently Ethereum Nodes quietly accept that.

None of these errors, except nonce-too-low, are realistic WalletKit errors.  But injecting them was useful for 'WK -> Blockset -> WK' end-to-end testing.

The PR is for a merge to `develop`.  Likely that `develop` does not make it to `master`.  Sigh.

Requires minimal iOS/Android changes for the new SubmitError, IncludeError, etc.  All compiler flagged.

**Review by DrewCarlson — APPROVED** [review]

**Review by kizoojun01 — COMMENTED** [review]
Bitcoin key

**Pull request merged**